lib/alchemy/resources_helper.rb

```diff
@@ -158,5 +158,15 @@ def render_resources
     rescue ActionView::MissingTemplate
       render :partial => 'resource', :collection => resources_instance_variable
     end
+
+    # Returns all the params necessary to get you back from where you where
+    # before: the Ransack query and the current page.
+    #
+    def current_location_params
+      {
+        q: params[:q],
+        page: params[:page]
```

Comment:
The problem with keeping the `page` parameter is, that the items on the search result page change, if one for instance, deletes a record. So, the page the user is being redirect to is wrong or even worse don't exist anymore. Let's keep this, but be aware, that your customer will complain about wrong result sets ;)